manifests/init.pp

```diff
@@ -53,6 +53,10 @@
   $path = $::path,
 ) {
 
+  include ::archive::params
+  Class['archive::params'] ->
+  Archive[$name]
+
```

Comment:
same as above, maybe that should also changed to a require ::archive::params??
But as already said, I'm not 100% sure about the include semantics, so these could be sufficient as well.

this piece is a bit older, with some older 3.x (about a year ago) Puppet versions, and I guess, I added
that dependency because Puppet did not always enforce the right order of evaluation, but I might
remember wrongly.